### DISCUSSION TITLE
Disable automatic ruleset update.

### DISCUSSION THREAD

**Discussion by aznzakum** [wazuh/wazuh#17456] [Q&A] [answered]
Good morning all,

Is there an effective way to disable automatic updates from Wazuh that would override the ruleset folder?

I noticed that version 4.2+ introduces update_ruleset script is now deprecated and removed.

This is also shown in https://github.com/wazuh/wazuh/issues/6904

The official Wazuh documentation states:
https://documentation.wazuh.com/current/user-manual/ruleset/update.html

"Since v4.2, Wazuh is delivered with the latest ruleset on each release. Manual update is no longer necessary nor supported."

The question is which file in the Wazuh manager pulls the automatic update?  Which configuration file (or init) is telling Wazuh to check and pull data to update the new rulesets? 

Thank you!

**Comment by mdiego92** [comment]
Hello @aznzakum,

The documentation is referring to the rule upgrade that happens with each Wazuh Manager upgrade installation.
**Is there any specific ruleset you don't want to be modified once you perform an upgrade?**

It is possible to copy the ruleset to the custom ones and disable it from the `ossec.conf` file in the manager. This could also generate some errors with child rules or decoders that may depend on these rules, but we can try.

**Example: We will set up the Palo Alto rules (`0700-paloalto_rules.xml`) as custom rules.**

- Copy the rule file:

`cp /var/ossec/ruleset/rules/0700-paloalto_rules.xml /var/ossec/etc/rules/
`

- Edit the ruleset section from the `/var/ossec/etc/ossec.conf` file in the Wazuh Manager to exclude the default `0700-paloalto_rules.xml` file even after the upgrade: 

```
  <ruleset>
    <!-- Default ruleset -->
    <decoder_dir>ruleset/decoders</decoder_dir>
    <rule_dir>ruleset/rules</rule_dir>
    <rule_exclude>0215-policy_rules.xml</rule_exclude>
    <rule_exclude>0700-paloalto_rules.xml</rule_exclude>
    <list>etc/lists/audit-keys</list>
    <list>etc/lists/amazon/aws-eventnames</list>
    <list>etc/lists/security-eventchannel</list>

    <!-- User-defined ruleset -->
    <decoder_dir>etc/decoders</decoder_dir>
    <rule_dir>etc/rules</rule_dir>
  </ruleset>
```

- Perform any rule modifications you may need and then restart the Wazuh Manager:

`/var/ossec/bin/wazuh-control restart
`

- Verify if you receive any errors in the ossec.log file:

`cat /var/ossec/logs/ossec.log | grep -iE 'ERR|WARN'CRIT'
`


Please let me know if that worked for you. Looking forward to your response!

**Reply by aznzakum** [reply]
Thank you so much for this comprehensive response. 

The situation of my organization is currently this:

We use our own proprietary platform to push specific wazuh rules. In order for this to work, our platform takes in all the rules from var/ossec/ruleset/rules and generates rules accordingly.
When we make whitelisting changes, we like to edit each .xml directly (this could be several countless .xml rules) within the ruleset/rules.

While it is recommended to edit things from the var/ossec/etc/rules/local_rules.xml, our platform does not like the use of certain keywords such as "overwrite=yes". Essentially, we would like absolute control over the master rules folder in order for our platform to work.

The warning from wazuh however is that **"Inside the ruleset/ folder you will find all the common rules and decoders. All files inside this folder will be overwritten or modified in the Wazuh update process, so please do not edit files or add custom files in this folder."**

This was stated at the bottom of the documentation in reference: https://documentation.wazuh.com/current/user-manual/ruleset/getting-started.html

I guess one of our burning questions is, if all files inside this folder is overwritten or modified in a Wazuh update process, are we able to manually change a config file in Wazuh that permanently disables this update process? And where in Wazuh's folders/directories holds the configuration for the update? From a granular level, how is Wazuh pulling the update and from where (is it through a cronjob?)

Thank you!

**Reply by mdiego92** [reply, marked as answer]
Hello @aznzakum,

It is impossible to stop the rules and decoders from changing with the upgrade, as it's part of the installation of the binary file. One possible workaround is to back up the folders and revert the changes after the upgrade. But, as I previously said, it is not possible to set that up through a config file or cronjob.

Thank you very much for your patience.

**Reply by aznzakum** [reply]
Thank you for your patience and time in answering my question. On a similar question, if I cannot do the above following, is it possible to stop the Wazuh Upgrader? 

Essentially the general question now I'm thinking is, do I have controls over stopping the automatic Wazuh upgrade? If there is a new version upgrade for Wazuh, my understanding is that it will automatically upgrade my Wazuh manager. 

Example, if Wazuh is currently on version 4.4.3 and a new version comes out tomorrow for 4.4.4, Is there somewhere I can configure to stop 4.4.4 from automatically being pushed to my manager? 

Thank you!

**Reply by mdiego92** [reply]
Hello @aznzakum,

Wazuh won't be automatically upgraded unless you execute anything that might do that. I mean, there is no specific functionality for automatic upgrades.

It could be possible that Wazuh is getting automatically upgraded with the Linux environment upgrades. The best option to avoid this is disabling the repositories:

For yum:

```
sed -i "s/^enabled=1/enabled=0/" /etc/yum.repos.d/wazuh.repo
sed -i "s/^enabled=1/enabled=0/" /etc/yum.repos.d/elastic.repo
```

or

For apt:

```
sed -i "s/^deb/#deb/" /etc/apt/sources.list.d/wazuh.list
sed -i "s/^deb/#deb/" /etc/apt/sources.list.d/elastic-7.x.list
apt-get update
```

Hopefully, this is what you're looking for!
